apps/segment/src/modules/db/segment-apl-repository.ts

```diff
@@ -0,0 +1,134 @@
+import { AuthData } from "@saleor/app-sdk/APL";
+import { DeleteItemCommand, GetItemCommand, PutItemCommand, ScanCommand } from "dynamodb-toolbox";
+import { err, ok, ResultAsync } from "neverthrow";
+
+import { BaseError } from "@/errors";
+import { createLogger } from "@/logger";
+import { SegmentAPLEntityType, SegmentMainTable } from "@/modules/db/segment-main-table";
+
+import { SegmentAPLMapper } from "./segment-apl-mapper";
+
+export class SegmentAPLRepository {
+  private logger = createLogger("DynamoDBAPLRepository");
+
+  private segmentAPLMapper = new SegmentAPLMapper();
+
+  static ReadEntityError = BaseError.subclass("ReadEntityError");
+  static WriteEntityError = BaseError.subclass("WriteEntityError");
+  static DeleteEntityError = BaseError.subclass("DeleteEntityError");
+  static ScanEntityError = BaseError.subclass("ScanEntityError");
+
+  constructor(
+    private deps: {
+      segmentAPLEntity: SegmentAPLEntityType;
+    },
+  ) {}
+
+  async getEntry(args: { saleorApiUrl: string }) {
+    const getEntryResult = await ResultAsync.fromPromise(
+      this.deps.segmentAPLEntity
+        .build(GetItemCommand)
+        .key({
+          PK: SegmentMainTable.getAPLPrimaryKey({
+            saleorApiUrl: args.saleorApiUrl,
+          }),
+          SK: SegmentMainTable.getAPLSortKey(),
+        })
+        .send(),
+      (error) =>
+        new SegmentAPLRepository.ReadEntityError("Failed to read APL entity", { cause: error }),
+    );
+
+    if (getEntryResult.isErr()) {
+      this.logger.error("Error while reading APL entity from DynamoDB", {
+        error: getEntryResult.error,
+      });
+
+      return err(getEntryResult.error);
+    }
+
+    if (!getEntryResult.value.Item) {
+      this.logger.warn("APL entry not found", { args });
+
+      return err(new SegmentAPLRepository.ReadEntityError("APL entry not found"));
+    }
```

Comment:
you can also consider less verobse interface: `ResultAsync<AuthData | null, Error>` where we push null as a successful data flow, separated from error flow. I'm not highly opinionated here, but handling will be less verbose than instanceOf